actionpack/test/dispatch/cookies_test.rb

```diff
@@ -1235,6 +1240,39 @@ def test_cookies_are_not_cleared
     assert_equal "bar", @controller.encrypted_cookie
   end
 
+  def test_signed_cookie_with_expires_set_relatively
+    freeze_time do
+      cookies.signed[:user_name] = { value: "assain", expires: 2.hours }
+
+      travel 1.hour
+      assert_equal "assain", cookies.signed[:user_name]
+
+      travel 2.hours
+      assert_nil cookies.signed[:user_name]
+    end
+  end
+
+  def test_encrypted_cookie_with_expires_set_relatively
+    freeze_time do
+      cookies.encrypted[:user_name] = { value: "assain", expires: 2.hours }
+
+      travel 1.hour
+      assert_equal "assain", cookies.encrypted[:user_name]
+
+      travel 2.hours
+      assert_nil cookies.encrypted[:user_name]
+    end
+  end
+
+  def test_vanilla_cookie_with_expires_set_relatively
+    freeze_time do
+      get :cookie_expires_in_two_hours
+
+      expected_time = 2.hours.from_now.utc.rfc2822
+      assert_cookie_header "user_name=assain; path=/; expires=#{expected_time}"
+    end
```

Comment:
@kaspth :smile: 
Previously you'd admonished that interpolating was a bad idea. However, I couldn't figure out the best way to test this yet :sweat_smile: